### PR TITLE
Fix ExistValidator is broken for NOSQL

### PR DESCRIPTION
**Pull request by developeruz**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #15726

### CODE REVIEW

**Review by samdark — APPROVED** [review]

**Comment by klimov-paul** [comment]
This is not enough: same fix should be applied to `valueExists()`.
https://github.com/yiisoft/yii2/blob/master/framework/validators/ExistValidator.php#L256-L262

**Review by klimov-paul — CHANGES_REQUESTED** [review]

File: framework/validators/ExistValidator.php [hunk 1]
```diff
@@ -126,7 +126,7 @@ private function checkTargetRelationExistence($model, $attribute)
             $relationQuery->andWhere($this->filter);
         }
 
-        if ($this->forceMasterDb) {
+        if ($this->forceMasterDb && method_exists($model::getDb(), 'useMaster')) {
```
[the remark this answers is not in the document]
Comment (developeruz) [reply framework/validators/ExistValidator.php]: Changed

**Comment by klimov-paul** [comment]
CHANGELOG entry should be added as well.

**Review by SilverFire — APPROVED** [review]

**Comment by SilverFire** [comment]
Thank you

**Pull request merged**